Add AC/DC converters Recom RAC20-xxSK and RAC20-xxDK

Werni2A opened the pull request:
Adds the 20W AC/DC converter series RAC20.

[https://recom-power.com/pdf/Powerline_AC-DC/RAC20-K.pdf](https://recom-power.com/pdf/Powerline_AC-DC/RAC20-K.pdf)
[FP geometry](https://recom-power.com/pdf/Powerline_AC-DC/RAC20-K.pdf#page=6)

![image](https://user-images.githubusercontent.com/58232418/92312127-30d03e80-efbe-11ea-9404-4b6d854829b6.png)
![image](https://user-images.githubusercontent.com/58232418/92312131-3cbc0080-efbe-11ea-90f5-d543bdcaa6d2.png)



---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [x] Provide a URL to a datasheet for the footprint(s) you are contributing
- [x] An example screenshot image is very helpful 
- [x] If there are matching symbol or 3D model pull requests, provide link(s) as appropriate
- [x] Check the output of the Travis automated check scripts - fix any errors as required
- [x] Give a reason behind any intentional library convention rule violation.

---

Be patient, we maintainers are volunteers with limited time and need to check your contribution against the datasheet. You can speed up the process by providing all the necessary information (see above). And you can speed up the process even more by providing a dimensioned drawing of your contribution. A tutorial on how to do that is found here: https://forum.kicad.info/t/how-to-check-footprint-correctness/9279 (This is optional!)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=2265) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KiCad/kicad-footprints?pullRequest=2265) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KiCad/kicad-footprints?pullRequest=2265) it.</sub>

aris-kimi commented:
Hi @Werni2A , thank you for your contribution.
From the image shown i can sugget to clear that triangle shaped pin1 fab indicator. 1mm bevel at the top left corner is the appropriate way.

Werni2A commented:
Hi @aris-kimi , thanks for taking a look at the footprint. I think this was my first contribution to the library. From my current point of view there are a few aspects that I want to change.

The most important one is that this kind of packages are quite common therefore I will try to write a generator script for it. Until then I mark this PR as WIP.

Werni2A commented:
@aris-kimi I updated the footprint. It would be great if you could take a look at it :)

This time created with [footprint-generator](https://github.com/pointhi/kicad-footprint-generator/pull/609)

Werni2A commented:
Semms like you were faster with linking the according generator ;)

aris-kimi commented on Converter_ACDC.pretty/Converter_ACDC_Recom_RAC20-xxDK_THT.kicad_mod:
```diff
@@ -0,0 +1,41 @@
+  (fp_text user "" (at -3 0) (layer F.SilkS)
+    (effects (font (size 1 1) (thickness 0.15)))
+  )
+  (model ${KISYS3DMOD}/Converter_DCDC.3dshapes/Converter_ACDC_Recom_RAC20-xxDK_THT.wrl
```
Hi @evanshultz and @myfreescalewebpage , this 3D path is wrong, because the script used was about DC/DC converters, can AC/DC added at the scripts?
If not i will advice to manually edit..Same for the second part.
I haven't reviewed them yet..

Werni2A replied:
In my opinion it would be great to make the script more generic e.g. passing the library as a parameter. Also passing the drill diameter without any pad information would be nice as this dimensions can be calculated via the IPC standard and most datasheets don't specify the pads.

Franck78 commented:
While I'm at working on those converters, I can add the 3dfootprint for this RAC20 series.
Don't add them, now it's 5 minutes to do the addition for me

https://github.com/KiCad/kicad-packages3D/issues/600

Just, I don't really know if I put the 3dimages in my current https://github.com/KiCad/kicad-packages3D/pull/721 PR
which allready have one converter or try another PR with all images.
?

Franck78 commented:
![rac20](https://user-images.githubusercontent.com/1778884/92329309-f2db2500-f066-11ea-9a41-51d433d11af8.png)

[Converter_ACDC.3dshapes.zip](https://github.com/KiCad/kicad-footprints/files/5179888/Converter_ACDC.3dshapes.zip)

Werni2A commented:
@Franck78 great that you add 3D models, I personally have no experience in creating those models. :+1: 

I think it's best to create a new PR and link it to this one. When everything is in only one PR it slows the whole process down as every part has to be checked before it can be merged. But I'm no librarian so that's just the impression I got.

Franck78 commented:
have you tried the 3D on the footprint just to confirm it is good ?

Yes, with 5 or more repo to follow, it is rapidly becoming messy.
I have about 12 new models, they must accept that as one thing or two, not twelve .

Werni2A commented:
I haven't check the dimensions but it looks good from a "optical inspection" :)
![image](https://user-images.githubusercontent.com/58232418/92336347-f2ab4b80-f09f-11ea-9d51-1d2a92e474d3.png)

Franck78 commented:
on the other side, the white line must follow the body, no space, no hide.

myfreescalewebpage commented:
Closing/opening to refresh the Travis test.

myfreescalewebpage commented:
@Werni2A starting the review here.

For both footprints:
- [ ] The description is very long, is speaking about a script we don't know and the datasheet is indicated twice. To be cleanup.
- [ ] The diameter of the pad is a little bit small based on the other converters. I suggest diameter=2.4mm and hole=1.3mm.

Joel

Werni2A commented:
Hi @myfreescalewebpage thanks for taking a look at this PR and sorry for the work.

The [Converter_DCDC](https://github.com/pointhi/kicad-footprint-generator/tree/master/scripts/Converter_DCDC) generator is not very well suited for this footprints. As soon as https://github.com/pointhi/kicad-footprint-generator/pull/624 is merged I wanted to rework/replace Converter_DCDC with a more generic DIP package generator.

Until someone is terribly in need for this footprint I would like to keep it on hold until I wrote the generator.

myfreescalewebpage commented:
As you wish, but there are only few comments on the footprint and https://github.com/KiCad/kicad-symbols/pull/2692 is waiting for this to be merged, so I think it can be very nice to finish manually and update later with scripted version.
I let you decide what you want to do.
Joel

Werni2A commented:
@myfreescalewebpage your comment is a good point. I updated the footprints manually. There were also some empty "user text" in the footprint that I removed.

myfreescalewebpage commented:
Thanks, at the moment the footprints can be merged. Up to you for a very nice converter script (you're welcome to do that !!)
